src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -12858,9 +13027,21 @@ void Assembler::prefix(Address adr) {
       prefix(REX_X);
     }
   }
+  if (is_map1) emit_int8(0x0F);
```

Comment:
- First. What `is_map1` means? There is no explanation for this name. May be add comment somewhere in `assembler_x86.hpp` file or use more meaningful name.

- Second. You added one more byte `0x0F` for instructions even when extended registers are not used and APX is not enabled. Why?  You added it in several `prefix()` and `prefixq()` methods. It can lead to regression since code size will increase.

Reply:
The is_map1 bool indicates an x86 map1 instruction which, when 
legacy encoded, uses a 0x0F opcode prefix.  By specification, the
opcode prefix is omitted when using rex2 encoding in support
of APX extended GPRs.

I added this comment before the relevant prefix functions in assembler_x86.hpp.  Is this sufficient?

Reply:
Comment is good.

Actually you should have point me to code which shows that code generation did not change, we generated `0x0F` before:
```
  - prefix(src);
  - emit_int16(0x0F, 0x18);
  + prefix(src, true /* is_map1 */);
  + emit_int8(0x18);
```